fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1411,6 +1411,9 @@ def _gather_optim_state(
             buffer = None  # for sharded tensors
             singleton_buffer = None  # for singleton tensors
             for buffer_name, t in v.items():
+                if torch.is_tensor(t):
+                    t = t.cuda()
```

Comment:
use `.to(self.compute_device)` instead, since it's possible the user hasn't properly called `torch.cuda.set_device`